src/textual/widgets/_tree.py

```diff
@@ -560,13 +560,20 @@ def get_label_width(self, node: TreeNode[TreeDataType]) -> int:
         label = self.render_label(node, NULL_STYLE, NULL_STYLE)
         return label.cell_len
 
-    def clear(self) -> None:
-        """Clear all nodes under root."""
+    def clear(self, label: TextType | None = None, data: TreeDataType = ...) -> None:
```

Comment:
I get the limitation of not being able to clear the root and I think this is so "exceptional" that it should be made obvious from the name of the method, e.g., by renaming it to `clear_children` or `clear_under_root`.